articles/advanced-r-concepts/index.md

```diff
@@ -0,0 +1,333 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/advanced-r-concepts/
+title: Advanced Programming Concepts in R
+description:
+author: lalithnarayan-c
+date: 2020-10-15T00:00:00-11:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/advanced-r-concepts/hero.jpg
+    alt:
+---
+In the first article, we have covered the basics of the R programming language. In this article, we will look at some of the advanced aspects of the R programming language. Going through this article will enhance one's ability to implement Object-Oriented Programming(OOP) concepts and write modular code. We will cover the following topics in this article:
+<!--more-->
+
+1. Dataframes
+2. Typecasting
+3. File handling
+4. Working with inbuilt datasets
+5. Conditional Statements
+6. Loop Statements
+
+### Dataframes
+R is a programming language built for statistical analysis of large datasets. Emphasis on handling large datasets is laid via dataframes. Dataframes facilitate the process of storing large datasets in a local variable.
```

Comment:
You give an example of data frames below but no formal definition.
You can cite a formal definition like the one on Tutorialspoint: "A data frame is a table or a two-dimensional array-like structure in which each column contains values of one variable and each row contains one set of values from each column"

Note that data frames is 2 separate words => correct this throughout article.